src/extensions/extra/ImageUrlCodeHints/main.js

```diff
@@ -0,0 +1,874 @@
+                            if (entry._isDirectory) {
+                                entryStr += "/";
+                            }
+
+                            // code hints show the unencoded string so the
+                            // choices are easier to read.  The encoded string
+                            // will still be inserted into the editor.
+                            unfiltered.push(entryStr);
+                        }
+                    });
+
+                    self.cachedHints.unfiltered = unfiltered;
+                    self.cachedHints.query      = query;
+                    self.cachedHints.queryDir   = queryDir;
+                    self.cachedHints.docDir     = docDir;
+
+                    if (self.cachedHints.deferred.state() !== "rejected") {
+                        currentDeferred = self.cachedHints.deferred;
+
+                        // Since we've cached the results, the next call to _getUrlList should be synchronous.
+                        // If it isn't, we've got a problem and should reject both the current deferred
+                        // and any new deferred that got created on the call.
+                        syncResults = self._getUrlList(query);
+                        if (syncResults instanceof Array) {
+                            currentDeferred.resolveWith(self, [syncResults]);
+                        } else {
+                            if (currentDeferred && currentDeferred.state() === "pending") {
+                                currentDeferred.reject();
+                            }
+
+                            if (self.cachedHints.deferred &&
+                                    self.cachedHints.deferred.state() === "pending") {
+                                self.cachedHints.deferred.reject();
+                                self.cachedHints.deferred = null;
+                            }
+                        }
+                    }
+                }
+            });
+
+            return self.cachedHints.deferred;
+        }
+
+        // without these entries, typing "../" will not display entries for containing folder
+        if (queryUrl.filename === ".") {
+            result.push(queryDir + ".");
+        } else if (queryUrl.filename === "..") {
+            result.push(queryDir + "..");
+        }
+
+        // add file/folder entries
+        unfiltered.forEach(function (item) {
+            if(LanguageManager.getLanguageForPath(item).getId() === "image") {
+                result.push(item);
+            }
+        });
+
+        var highestNumber = 0;
+        result.forEach(function (item){
+            item = item.split("/");
+            item = item[item.length-1];
+            if(item.indexOf("selfie") !== -1 && item.indexOf("selfie") === 0) {
+                // Removes extension from filename
+                item = item.split(".")[0];
+
+                var currentNumber = Number(item.substr(6));
```

Comment:
This will break on the 10th item, and for every one thereafter.  A better way to do this would be to regex out the number portion, something like:

```
var filenameParts = /selfie(\d*)\.png/;
var currentNumber = filenameParts && filenameParts[1] ? Number(filenameParts) : 0;
```